packages/flutter/lib/src/material/carousel.dart

```diff
@@ -14,34 +14,75 @@ import 'ink_well.dart';
 import 'material.dart';
 import 'theme.dart';
 
+// Examples can assume:
+// late BuildContext context;
+
 /// A Material Design carousel widget.
 ///
 /// The [CarouselView] presents a scrollable list of items, each of which can dynamically
 /// change size based on the chosen layout.
 ///
-/// Material Design 3 introduces 4 carousel layouts:
+/// Material Design 3 introduced 4 carousel layouts:
 ///  * Multi-browse: This layout shows at least one large, medium, and small
-/// carousel item at a time.
-///  * Uncontained (default): This layout show items that scroll to the edge of
-/// the container.
+///    carousel item at a time. This layout is supported by [CarouselView.weighted].
+///  * Uncontained (default): This layout show items that scroll to the edge of the
+///    container. This layout is supported by [CarouselView].
 ///  * Hero: This layout shows at least one large and one small item at a time.
+///    This layout is supported by [CarouselView.weighted].
 ///  * Full-screen: This layout shows one edge-to-edge large item at a time and
-/// scrolls vertically.
+///    scrolls vertically. The full-screen layout can be supported by both
+///    constructors.
+///
+/// The default constructor implements the uncontained layout model. It shows
+/// items that scroll to the edge of the container, behaving similarly to a
+/// [ListView] where all children are a uniform size. [CarouselView.weighted]
+/// enables dynamic item sizing. Each item is assigned a weight that determines
+/// the portion of the viewport it occupies. This constructor helps to create
+/// layouts like multi-browse, and hero. In order to have a full-screen layout,
+/// if [CarouselView] is used, then set the [itemExtent] to screen size; if
+/// [CarouselView.weighted] is used, then set the [flexWeights] to only have
+/// one integer in the array.
+///
+/// {@tool snippet}
 ///
-/// This widget supports uncontained carousel layout. It shows items that scroll
-/// to the edge of the container, behaving similarly to a [ListView] where all
-/// children are a uniform size. [CarouselView.weighted] enables dynamic item
-/// sizing. Each item is assigned a weight that determines the portion of the
-/// viewport it occupies. This allows you to easily create layouts like
-/// multi-browse, hero, and full-screen.
+/// This code snippet shows how to get a vertical full-screen carousel by using
+/// [itemExtent] in [CarouselView].
+///
+/// ```dart
+/// Scaffold(
+///   body: CarouselView(
+///     scrollDirection: Axis.vertical,
+///     itemExtent: MediaQuery.sizeOf(context).height,
```

Comment:
This would be the size of the device screen, not necessarily the size of the CarouselView. What if the carousel is smaller than the device screen?

Reply:
To make the `itemExtent` match the size of the carousel, we can use `double.infinity`. Just updated the sample code:)!